app/scripts/lib/PatchStore.ts

```diff
@@ -14,9 +15,9 @@ export class PatchStore {
   private pendingPatches: Map<string, Patch> = new Map();
 
   private listener: (request: {
-    controllerKey: string;
-    oldState: Record<string, unknown>;
-    newState: Record<string, unknown>;
+    controllerKey?: keyof BackgroundStateProxy;
+    oldState: BackgroundStateProxy;
```

Comment:
As mentioned in other comments, this and `newState` isn't the full background state, but just the state for a single controller.

Reply:
Thanks for pointing this out! I've updated `PatchStore` so that it only handles individual controller state, with accompanying updates to `sanitizeUIState`.

I also realized I had missed some links in the data flow between `MetamaskController` and the Redux store. I updated the `updateMetamaskState` and `applyPatches` Redux actions accordingly.

These are the diffs for the new changes: https://github.com/MetaMask/metamask-extension/compare/27f17e3a4729a1a868934548fc086662e87d2d91..d83796f39a18e0e3b09a29c0f96d4278f48ce6fc